system/View/Cells/Cell.php

```diff
@@ -64,37 +65,48 @@ public function setView(string $view)
      * from within the view, this method extracts $data into the
      * current scope and captures the output buffer instead of
      * relying on the view service.
+     *
+     * @throws RuntimeException
      */
     final protected function view(?string $view, array $data = []): string
     {
         $properties = $this->getPublicProperties();
         $properties = $this->includeComputedProperties($properties);
         $properties = array_merge($properties, $data);
 
-        // If no view is specified, we'll try to guess it based on the class name.
-        if (empty($view)) {
-            // According to the docs, the name of the view file should be the
-            // snake_cased version of the cell's class name, but for backward
-            // compatibility, the name also accepts '_cell' being omitted.
-            $ref      = new ReflectionClass($this);
-            $view     = decamelize($ref->getShortName());
-            $viewPath = dirname($ref->getFileName()) . DIRECTORY_SEPARATOR . $view . '.php';
-            $view     = is_file($viewPath) ? $viewPath : str_replace('_cell', '', $view);
+        $view = (string) $view;
+
+        if ($view === '') {
+            $viewName  = decamelize(class_basename(static::class));
+            $directory = dirname((new ReflectionClass($this))->getFileName()) . DIRECTORY_SEPARATOR;
+
+            $possibleView1 = $directory . substr($viewName, 0, strrpos($viewName, '_cell')) . '.php';
+            $possibleView2 = $directory . $viewName . '.php';
+        }
+
+        if ($view !== '' && ! is_file($view)) {
+            $directory = dirname((new ReflectionClass($this))->getFileName()) . DIRECTORY_SEPARATOR;
+
+            $view = $directory . $view . '.php';
+        }
+
+        foreach ([$view, $possibleView1 ?? '', $possibleView2 ?? ''] as $candidateView) {
+            if (is_file($candidateView)) {
+                $foundView = $candidateView;
+                break;
+            }
         }
 
-        // Locate our view, preferring the directory of the class.
-        if (! is_file($view)) {
-            // Get the local pathname of the Cell
-            $ref  = new ReflectionClass($this);
-            $view = dirname($ref->getFileName()) . DIRECTORY_SEPARATOR . $view . '.php';
+        if (! isset($foundView)) {
+            throw new RuntimeException('Cannot locate the view file for the cell.');
```

Comment:
Is this a runtime exception?
It seems to me that it is a logic exception.

> LogicException
Exception that represents error in the program logic. This kind of exception should lead directly to a fix in your code. 
https://www.php.net/manual/en/class.logicexception.php